### PR TITLE
Add option 'enable_pass' which can override making the 'Set DCO to pa…

### PR DESCRIPTION
**Pull request by jmertic**
…ss' option for those with write access

Signed-off-by: John Mertic <jmertic@linuxfoundation.org>

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -61,60 +64,93 @@ module.exports = app => {
...
-      status: 'completed',
-      started_at: timeStart,
-      conclusion: 'success',
-      completed_at: new Date(),
-      output: {
-        title: 'DCO',
-        summary: 'Commit sign-off was manually approved.'
-      }
-    }))
+    if (enablePass) {
```
Comment (hiimbex) [inline index.js]: `enablePass` was a variable you set in the context of the function `check`. `check` happens when commits are pushed and is based on webhook events. No state is maintained from there when the `check_run.requested_action` event is called, triggering this `setStatusPass` function.

You'll need to add something like https://github.com/probot/dco/pull/120/files#diff-168726dbe96b3ce427e7fedce31bb0bcR11 and https://github.com/probot/dco/pull/120/files#diff-168726dbe96b3ce427e7fedce31bb0bcR18 in this function.

Comment (jmertic) [reply index.js]: Good catch! Fixed now :-).

Comment (jmertic) [reply index.js]: Fixed the build error - please review

Comment (jmertic) [reply index.js]: @hiimbex - Can you review?

Comment (jmertic) [reply index.js]: @hiimbex can you review?

Comment (jmertic) [reply index.js:145]: Adding @brianwarner who also offered to review.

**Comment by jmertic** [comment]
@hiimbex @brianwarner Can you take a look and see what else I need to do here?

**Comment by gr2m** [comment]
Hey @jmertic maybe I can help.

I don't see an associated issue for `enable_pass` and no documentation update in this PR, can you elaborate what it is for exactly? Is it to disable the "Set DCO to pass" check run action?

**Comment by jmertic** [comment]
Hey @gr2m - thanks for jumping in here.

Related issue is at https://github.com/probot/dco/issues/119. You are right that this is just to disable the "Set DCO to pass" check run action, as it makes the commit history paper trail hard to follow for DCO signoffs missed.

I'm happy to add any documentation or anything else needed to move this along.

**Comment by gr2m** [comment]
sounds good, thank you for the clarification.

Can you rebase your changes on the latest in `master`? Can you add the option to the README, and also add an integration test for the behavior so we won't break it in the future?

**Comment by jmertic** [comment]
Thanks - it's on my list. Will try to get this update in the next week or so.

**Comment by fabianofernandeszup** [comment]
Was this PR abandoned?

**Comment by jmertic** [comment]
It hasn't - I've just been swamped to have time and do this. Sounds like the main pieces are docs and tests, which I'll try to wrap up soon. Thanks for the reminder!

**Review by reedhhw — APPROVED** [review]

**Review by reedhhw — APPROVED** [review]

**Review by reedhhw — APPROVED** [review]

**Review by reedhhw — APPROVED** [review]

**Review by reedhhw — APPROVED** [review]

**Comment by jmertic** [comment]
@gr2m @hiimbex I've got this all done and documented, but I'm struggling on a test. Is there an example of how to create an option in the `dco.yml` file to test this?

**Comment by hiimbex** [comment]
there's 2 main tests in this repo and they have very different methodology. 

The [first](https://github.com/probot/dco/blob/master/test/index.test.js) uses github api mocks, so you'd mock the return of fetching the config files content from the github api (specifically this: https://developer.github.com/v3/repos/contents/#get-repository-content) - I think it uses https://github.com/probot/octokit-plugin-config now. Then you can basically verify whether or not the `actions` block was passed into the create check api call.

The second uses invocations of the `getDCOStatus` function with various arguments which your change doesnt really touch.

File: index.js [hunk 2]
```diff
@@ -86,69 +88,80 @@ module.exports = (app) => {
...
-            output: {
-              title: 'DCO',
-              summary
-            },
-            actions: [
-              {
-                label: 'Set DCO to pass',
-                description: 'would set status to passing',
-                identifier: 'override'
+      if (enablePass) {
```
Comment (hiimbex) [inline index.js:91]: this if statement makes sense, but there's no `else` later on. If you don't have an else and include an api call without the `actions` block, no failing check will get created at all

Comment (hiimbex) [reply index.js:91]: actions block referring to https://github.com/probot/dco/pull/120/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R106-R112

Comment (jmertic) [reply index.js:91]: Good point! Added that now - let me know if the status returned makes sense.

**Comment by jmertic** [comment]
@himbex - Thanks I will take a look at those examples for writing the test.

**Comment by jmertic** [comment]
@hiimbex Hey - still looking at this but being not a node person I'm having trouble wrapping my head around the test. Could you help perhaps?

**Review by jasawan — COMMENTED** [review]
[Add option 'enable_pass' which can override making the 'Set DCO to pa… #120](https://github.com/probot/dco/pull/120/files/3ba5e9bd58502c6f84f0d031b75b859105a36d32)
